.github/workflows/ci-pipeline.yaml

```diff
@@ -40,11 +56,11 @@ jobs:
   tests:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
For now not including the `harden-runner` step here, because it causes the [tests to fail](https://github.com/Azure/aks-periscope/actions/runs/7777037330).

This seems to be due to some DNS queries for local (`*.svc.cluster.local`) domains within the `kind` cluster, which the harden-runner agent attempts to resolve externally.